packages/samples/vuejs/README.md

```diff
@@ -1,19 +1,231 @@
+createApp(App).mount('#app');
 ```
 
-### Compiles and minifies for production
+## Using the CDN
+
+Follow the instructions [here](https://docs.coveo.com/en/atomic/latest/usage/#cdn) to add the CDN script to your entry html file. The script will initialize your components, avoiding the need to manually do it like in the npm package usage above. You will also not need to copy any assets over.
+
+## Initializing your search interface
+
+After you render your search interface, make sure to initialize it
+
+```js
+import {onMounted} from 'vue';
+
+async function initInterface() {
+  await customElements.whenDefined('atomic-search-interface');
+  const searchInterface = document.querySelector(
+    'atomic-search-interface'
+  ) as HTMLAtomicSearchInterfaceElement;
+
+  // Initialization
+  await searchInterface.initialize({
+    accessToken: '<YOUR-TOKEN>',
+    organizationId: '<YOUR-ORGANIZATION-ID>',
+  });
+
+  // Trigger a first search
+  searchInterface.executeFirstSearch();
+}
+
+onMounted(initInterface);
 ```
-npm run build
+
+### Using `<atomic-result-template>`
+
+As you are aware, Vue treats the `<template>` tag as a container for Vue components and therefore doesn't render it to the page. This poses an obstacle for working with the `<atomic-result-template>`, which expects a native `<template>` that it will then use to create each result element. In order for this to work, your result templates will need to live in an html file, like in this example's `src/templates/result-template.html` and then imported and used like below
+
+```html
+<!-- .vue component -->
+<script setup lang="ts">
+import resultTemplate from '../templates/result-template.html';
+</script>
+
+<template>
+    <atomic-result-list>
+      <atomic-result-template v-html="resultTemplate"></atomic-result-template>
+    </atomic-result-list>
+</template>
+
 ```
 
-### Customize configuration
-See [Configuration Reference](https://cli.vuejs.org/config/).
+### Using custom components
+
+Writing your own components to leverage Vue components should be as straightforward as using the atomic components on any other page. However, it becomes a little bit trickier when you want to use a custom Vue component inside your result templates which, as shown above, need to be passed down as raw html.
+
+For example, in this project we defined the simple `ResultTextField` component, which takes a `label` and a `field` as `props` and render some atomic components
+
+```html
+<!-- ResultTextField.vue -->
+<script setup lang="ts">
+const props = defineProps({
+  label: String,
+  field: String,
+});
+</script>
+
+<template>
+  <atomic-text :value="props.label"></atomic-text>:
+  <atomic-result-text :field="props.field"></atomic-result-text>
+</template>
+
+```
+
+If you try to use this component in an html template like the snippet below and then try to use in in the component below it, Vue will not attempt to render it, as it expects onli native html to be passed via the `v-html` directive.
```

Comment:
```suggestion
If you try to use this component in an html template like the snippet below, and then try to use it in the component further below it, Vue will not attempt to render it. The reason is that Vue it expects only native HTML to be passed via the `v-html` directive.
```
(maintly just because the sentence was long and harder to read for that reason)